browser/lib/util/crypto.js

```diff
@@ -26,7 +26,8 @@ var Crypto = (function() {
 			Logger.logAction(Logger.LOG_MAJOR, 'Ably.Crypto.generateRandom()', 'Warning: the browser you are using does not support secure cryptographically secure randomness generation; falling back to insecure Math.random()');
 			var words = bytes / 4, array = new Array(words);
 			for(var i = 0; i < words; i++) {
-				array[i] = Math.floor(Math.random() * VAL32);
+				/* subtract VAL32/2 as cryptojs wordarrays use signed ints */
+				array[i] = Math.floor(Math.random() * VAL32) - (VAL32 / 2);
```

Comment:
Sure, though I'm assuming you mean `VAL31` -- If `VAL32` is 2<sup>32</sup>, I'd expect a constant called `VAL16` to be 2<sup>16</sup>, not 2<sup>31</sup>

or better still, maybe `UINT32_SUP` and `INT32_SUP` respectively